src/main/java/seedu/address/logic/filters/CarFilter.java

```diff
@@ -42,19 +46,20 @@ public static Car parseCar(String car) {
     /**
      * Predicate for abstract test method use.
      *
-     * @param car Car member from carsOwned or carsPreferred.
+     * @param car       Car member from carsOwned or carsPreferred.
      * @param filterCar Specified Car to filter by.
      * @return
      */
-    public boolean carFilterPredicate(Car car, Car filterCar) {
-        if (filterCar.carBrand == null) {
-            return car.isSameCarType(filterCar);
-        } else if (filterCar.carType == null) {
-            return car.isSameCarBrand(filterCar);
-        } else if ((filterCar.carBrand != null) && (filterCar.carType != null)) {
-            return car.isSameCarBrandAndCarType(filterCar);
+    public boolean carFilterPredicate(Car car, String filterCar) {
+        filterCar = filterCar.trim();
+        String[] carDetails = filterCar.split("\\+", 2);
+        if (carDetails.length == 1) {
+
+            return car.getCarBrand().toLowerCase().contains(filterCar.toLowerCase())
+                || car.getCarType().toLowerCase().contains(filterCar.toLowerCase());
         } else {
-            return false;
+            return car.getCarBrand().toLowerCase().contains(carDetails[0].toLowerCase())
+                || car.getCarType().toLowerCase().contains(carDetails[1].toLowerCase());
```

Comment:
Did you OK this with the writer of CarFilter? Cause this is probably different from the intent of @OhJunMing